miden-lib/asm/scripts/ASWAP.masm

```diff
@@ -0,0 +1,53 @@
+use.miden::sat::account
+use.miden::sat::note
+use.miden::sat::tx
+
+# Atomic Swap script: adds an asset from the note into consumers account and
+# creates a note consumable by initial issuer containing requested ASSET.
+#
+# Requires that the account exposes: 
+#
+# Inputs: []
+# Outputs: []
+#
+# Note inputs are assumed to be as follows:
+#
+# FAILS if:
+begin
+    # Process:
+    # 1. Alice creates a note A containing ASSET X consumable by anyone
+    # 2. Bob consumes note A receiving ASSET X
+    # 3. Bob create a note B containing ASSET Y
+    # 
+    # Requirements:
+    # - Anyone can consume note A
+    # - Note B needs to contain ASSET Y initialy requested by Alice
+    # - Only Alice can consume note B
+
+    # drop the transaction script root
+    dropw
+    # => []
+
+    # load the note inputs to memory starting at address 0
+    push.0 exec.note::get_inputs
+    # => [inputs_ptr] 
+    
+    # load the asset into the stack
+    exec.get_assets drop mem_loadw
```

Comment:
I would start with this and get the asset first into the account. 

So it would look like this 

```
use.miden::sat::account
use.miden::sat::note
use.miden::sat::tx
use.miden::wallets::basic->wallet

begin
    # drop the transaction script root
    dropw
    # => [...]
    
    # add the asset to the account 
    exec.note::get_assets mem_loadw call.wallet::receive_asset dropw
    # => [...]
    
    # then load the note inputs ASSET, TAG, RECIPIENT as Bobbin suggests
    note::get_inputs
    # => [inputs_ptr] - now your inputs are on mem0, mem1, ...

    # Now do some manipulation such that you have what you need 
    # ASSET, tag, RECIPIENT (whereas when capitalized we always mean a Word, and non-capitalized, we mean a Felt)

    # Now create the new note (I would use the wallet procedures for simplicity)
    call.wallet::receive_asset

end
``` suggestion